Add the debug option and verbose (foreground)

folker-kuhn opened the pull request:
Fixes #1954

I am proposing this PR, so we can debug and do foreground run (or start?) easily. I took "suspend" and "verbose" names from some other projects, so I think the user is familiar with it.

folker-kuhn committed:
Add the debug option and verbose (foreground)

I am proposing this PR, so we can debug and do foreground run (or start?) easily. I took "suspend" and "verbose" names from some other projects, so I think the user is familiar with it.

Thihup commented:
Hi @folker-kuhn.

As this is your first PR can you please acknowledge that the code will be owned by Manorrock.com. I will only ask this one time, as that is all that is needed for contributing code back to us.

About running in foreground, the `run.sh` does exactly that, so that it wouldn't be needed.
Also, could you implement the suspend part it in the `run.sh` script too?

folker-kuhn commented:
I acknowledge the code in this PR and all further PRs I do to this project will be owned by Manorrock.com. Thanks!

Okay, that is what `run.sh` is for. I have to say that having both `start` and `run` is somewhat confusing. If you must have separate scripts for this, maybe `start-bg` and `start-fg` would be easier to grasp.

I'll look at `run.sh`

mnriem commented:
@folker-kuhn Thanks for the acknowledgement. I'll let @Thihup work with you on the PR as he already engaged. We can now start merging any of your contributions. Welcome and thank you!

folker-kuhn committed:
Merge branch 'piranhacloud:current' into patch-1

folker-kuhn committed:
Update start.sh

Adding --run option to start in background with a wait

folker-kuhn committed:
Update run.sh

Make sure all options in start.sh also apply when using run.sh

folker-kuhn committed:
Update start.sh

Touch the pid before the command, since there is some loop looking for it in the cmd, and the pid is otherwise created after it. We get a race here.

arjantijms reviewed (approved):
LGTM

Thihup commented on server/src/main/zip/bin/run.sh:
```diff
@@ -1,23 +1,3 @@
-PID=$!
-echo $PID >> tmp/piranha.pid
-wait $PID
+exec ./start.sh --run
```
```suggestion
exec ./start.sh --run $*
```
Adding the `$*` allows to `./run.sh --suspend`

folker-kuhn replied:
I like that idea, thanks!

folker-kuhn committed:
Update server/src/main/zip/bin/run.sh

Co-authored-by: Thiago Henrique Hüpner <thihup@gmail.com>

Thihup committed:
Merge branch 'current' into patch-1

folker-kuhn commented:
@Thihup what happened with the last commit and test run?

folker-kuhn commented:
Error:  Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.3.0:jar (attach-javadocs) on project piranha-pico-core: MavenReportException: Error while generating Javadoc: 
Error:  Exit code: 1 - Loading source file /home/runner/work/piranha/piranha/pico/core/src/main/java/module-info.java...
Error:  Constructing Javadoc information...
Error:  error: No public or protected classes found to document.
Error:  1 error

I don't think this is related to my commit?

Thihup commented:
@folker-kuhn It looks like @mnriem added a new subproject. However, it does not have any classes yet. So the Javadoc is complaining about it. It is not related to your changes. Sorry for the confusion.

arjantijms committed:
Merge branch 'current' into patch-1

mnriem commented:
@folker-kuhn Thank you for your first contribution! It is MUCH appreciated and sorry for the confusion with the build ;)
